raiden/ui/startup.py

```diff
@@ -33,6 +48,107 @@
+                f"{to_checksum_address(self.secret_registry.address)}"
+            )
+
+
+@dataclass(frozen=True)
+class ServicesBundle:
+    user_deposit: UserDeposit
+    service_registry: Optional[ServiceRegistry]
+    monitoring_service: Optional[MonitoringService]
+    one_to_n: Optional[OneToN]
+
+    def __post_init__(self) -> None:
+        block_identifier = "latest"
+        user_deposit_address = self.user_deposit.address
+        token_address = self.user_deposit.token_address(block_identifier)
+
+        monitoring_service_address = self.user_deposit.monitoring_service_address(block_identifier)
+
+        # Validation should only be done if monitoring is enabled or PFS is used
+        if (
+            self.monitoring_service is None
+            or self.service_registry is None
+            or self.one_to_n is None
+        ):
+            return
+
+        if monitoring_service_address != self.monitoring_service.address:
+            click.secho(
+                f"Monitoring service address linked with the user deposit contract "
+                f"{to_checksum_address(monitoring_service_address)} does not match "
+                f"the address provided by the monitoring service proxy "
+                f"{to_checksum_address(self.monitoring_service.address)}"
+            )
+        one_to_n_address = self.user_deposit.one_to_n_address(block_identifier)
+        if one_to_n_address != self.one_to_n.address:
+            click.secho(
+                f"OneToN address linked with the user deposit contract "
+                f"{to_checksum_address(one_to_n_address)} does not match "
+                f"the address provided by the OneToN proxy "
+                f"{to_checksum_address(self.one_to_n.address)}"
+            )
+        service_registry_address = self.monitoring_service.service_registry_address(
+            block_identifier
+        )
+        if service_registry_address != self.service_registry.address:
+            click.secho(
+                f"The service registry address linked with the monitoring service contract "
+                f"{to_checksum_address(service_registry_address)} does not match "
+                f"the address provided by the service registry proxy "
+                f"{to_checksum_address(self.service_registry.address)}"
+            )
+
+        token_address_matches_monitoring_service = (
+            token_address == self.monitoring_service.token_address(block_identifier)
+        )
+        if not token_address_matches_monitoring_service:
+            msg = (
+                f"The token used in the provided user deposit contract "
+                f"{user_deposit_address} does not match the one in the "
+                f"MonitoringService contract {monitoring_service_address}."
+            )
+            click.secho(msg, fg="red")
+            sys.exit(1)
+
+        token_address_matches_one_to_n = token_address == self.one_to_n.token_address(
```

Comment:
What is the point of this check? The `one_to_n` contract does not have a `token_address`. According to this PR there is a `token_address` function for the `one_to_n` proxy, but that function does not work.